Sources/NIO/Bootstrap.swift

```diff
@@ -967,3 +978,54 @@ public final class NIOPipeBootstrap {
         }
     }
 }
+
+/// An channel option which can be applied to bootstrap using shorthand notation.
+/// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+public struct ShorthandBootstrapOption {
+    private let apply: ShorthandApply
+    
+    private init(apply: ShorthandApply) {
+        self.apply = apply
+    }
+    
+    /// Apply the contained option to the supplied ServerBootstrap
+    /// - Parameter serverBootstrap: bootstrap to apply this option to.
+    /// - Returns: the modified bootstrap (currently the same one mutated)
+    func applyOption<BootstrapT : ServerBootstrap>(serverBootstrap : BootstrapT) -> BootstrapT {
+        return apply.applyOption(serverBootstrap: serverBootstrap)
+    }
+}
+
+private protocol ShorthandApply {
+    func applyOption<BootstrapT : ServerBootstrap>(serverBootstrap : BootstrapT) -> BootstrapT
+}
+
+private struct ShorthandApplyImpl<Option : ChannelOption> : ShorthandApply {
+    let option : Option
+    let value : Option.Value
+    
+    func applyOption<BootstrapT : ServerBootstrap>(serverBootstrap: BootstrapT) -> BootstrapT {
+        return serverBootstrap.serverChannelOption(option, value: value)
+    }
+}
+
+/// Approved shorthand options.
+extension ShorthandBootstrapOption {
+    /// Option to reuse address.
+    /// - See:  NIOBSDSocket.Option.reuseaddr
+    public static let reuseAddr = ShorthandBootstrapOption(apply: ShorthandApplyImpl(option: ChannelOptions.socketOption(.reuseaddr), value: 1))
```

Comment:
we should probably give this a nice name. Like `allowImmediateAddressReuse` or something. Subject to bike-shedding

Reply:
If we could go back in time yes, however now I suspect most people using this will understand reuseAddr - probably just adding confusion to give it a "nice name"

Reply:
Done.